cli/cmd/cluster_gcp.go

```diff
@@ -449,34 +454,56 @@ func createGKECluster(clusterConfig *clusterconfig.GCPConfig, gcpClient *gcp.Cli
 				},
 				InitialNodeCount: 1,
 			},
-			{
-				Name: "ng-cortex-worker-on-demand",
-				Config: &containerpb.NodeConfig{
-					MachineType: *clusterConfig.InstanceType,
-					Labels:      nodeLabels,
-					Taints: []*containerpb.NodeTaint{
-						{
-							Key:    "workload",
-							Value:  "true",
-							Effect: containerpb.NodeTaint_NO_SCHEDULE,
-						},
+		},
+		Locations: []string{*clusterConfig.Zone},
+	}
+
+	if *clusterConfig.Preemptible {
+		gkeClusterConfig.NodePools = append(gkeClusterConfig.NodePools, &containerpb.NodePool{
+			Name: "ng-cortex-wk-preemp",
+			Config: &containerpb.NodeConfig{
+				MachineType: *clusterConfig.InstanceType,
+				Labels:      nodeLabels,
+				Taints: []*containerpb.NodeTaint{
+					{
+						Key:    "workload",
+						Value:  "true",
+						Effect: containerpb.NodeTaint_NO_SCHEDULE,
 					},
-					Accelerators: accelerators,
-					OauthScopes: []string{
-						"https://www.googleapis.com/auth/compute",
-						"https://www.googleapis.com/auth/devstorage.read_only",
+				},
+				Accelerators: accelerators,
+				OauthScopes: []string{
+					"https://www.googleapis.com/auth/compute",
+					"https://www.googleapis.com/auth/devstorage.read_only",
+				},
+				ServiceAccount: gcpClient.ClientEmail,
+				Preemptible:    true,
+			},
+			InitialNodeCount: int32(initialNodeCount),
```

Comment:
Could it make sense for this to be 0 if on-demand backup is used? Just a thought I had, we can discuss it live